### ISSUE TITLE
Autocomplete for JS files but not TS files? (TypeScript 2.2 and 2.3)

### ISSUE THREAD

**Issue by Meligy** [Microsoft/vscode#24956]
- VSCode Version:

  VS Code Insiders
  Version 1.12.0-insider (1.12.0-insider)

- OS Version:

  macOS Sierra
  10.12.4 (16E195)

Steps to Reproduce:

```
mkdir test-dir
cd test-dir
npm i -D typescript
node_modules/.bin/tsc --init
touch index.ts
touch index.js
npm i snoowrap # Any library without types
code-insiders .
```

Now, Look at this, in `index.js`:

![image](https://cloud.githubusercontent.com/assets/171637/25134085/00eafd40-2492-11e7-8c14-113dd9312ede.png)

However, in `index.ts`:

![image](https://cloud.githubusercontent.com/assets/171637/25134136/249f58e4-2492-11e7-8194-e81f517b0247.png)

I tried a few libraries, like `reddit-oauth`, and `chrome-remote-interface` (sorry for the obscure choices, it wasn't the intention of testing this). 

Basically any library without typings. Libraries with typings, as well as my own files, all work very well. For example, all my Angular 2 applications) work just fine.

The particular pictures show no suggestions because I also disabled snippets `"editor.snippetSuggestions": "none"`, but I confirmed with and without this setting.

I tried this with both TypeScript 2.2 and 2.3. I tried messing with several options in `tsconfig.json`, and it didn't seem to make a difference. 
Things like `"allowJs": true,` or enabling `typeAcquisition` (not sure what it is). I also had `"moduleResolution": "node",` all the time.

**Comment by mjbvz** [comment]
Yes, for TypeScript code, any library you use must have typings, either from `@types/...` or  included in the package. It does not look like there are typings for `snoowrap`. 

In TypeScript projects, these typings files must be manually managed. In JavaScript, since the Typings files are only used for editor language features and not for compiling, we will automatically download these typings to provide richer intellisense. Here's some more info about automatic typing acquisition: https://code.visualstudio.com/Docs/languages/javascript#_intellisense

In your case, you have a few options to work with this library:

* Write a `*.d.ts` typings defintion file for the module. Here's some starting points for that: http://stackoverflow.com/questions/12687779/how-do-you-produce-a-d-ts-typings-definition-file-from-an-existing-javascript
* Pretend you are working with JavaScript by casting `snoowrap` to an `any` type. This disables all type checking

Please let me know if that makes sense or if you have questions about this

**Comment by Meligy** [comment]
Hi Matt,

The reason I'd expect this to work is the [Untyped imports](https://github.com/Microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#untyped-imports) feature [announced in TypeScript 2.1](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-1.html#untyped-imports).

If there are no typings at all for JS, VS Code seems to generate some typings or use another technique to understand the structure of the non typed module.

In this particular case I'm experimenting with a new JavaScript-only library, and I'm appreciating that VS Code can get as close as possible to TypeScript when working with it. I thought it definitely would use that support for untyped imports, given they're supported TypeScript feature.

**Comment by mjbvz** [comment]
Ok. Thanks for the clarification.

In the untyped import case, TypeScript essentially ends up treating the imported module as an `any` type. This is why IntelliSense no intellisense shows on the module.  However I do agree that it makes sense to show some suggestions on it since we do in the JavaScript case. I've opened https://github.com/Microsoft/TypeScript/issues/15286 to track this

**Comment by Meligy** [comment]
Thanks a lot :)
